skorch/tests/test_net.py

```diff
@@ -2490,6 +2490,36 @@ def initialize_module(self, *args, **kwargs):
         hidden_units = net.mymodule_.state_dict()['sequential.3.weight'].shape[1]
         assert hidden_units == 99
 
+    @pytest.mark.parametrize("needs_y, train_split, raises", [
+        (False, None, ExitStack()),  # ExitStack = does not raise
+        (True, None, ExitStack()),
+        (False, lambda x: (x, x), ExitStack()),
+        (True, lambda x, y: (x, x), ExitStack()),
+        (True, lambda x: (x, x), pytest.raises(TypeError)),  # Raises an error
+    ])
+    def test_train_split_with_nans(self, needs_y, train_split, raises):
```

Comment:
I'm a bit confused about the name `test_train_split_with_nans`, where do nans come into play? Is the test not about `y` being passed to `train_split` only when it is not None?

Reply:
General remark: indeed, I didn't think well about the name of the test case. 

> where do nans come into play? 

The `NaN`s here come into play as `y=None` and `train_split=None`

>  Is the test not about y being passed to train_split only when it is not None?

I am writing a more verbose answer to this question as the PR comment, so we can move the discussion there

Reply:
So, can we call this function `test_passes_y_to_train_split_when_not_nan`? Frankly, I can't invent a better name 😅

Reply:
> The `NaN`s here come into play as `y=None` and `train_split=None`

Can you elaborate on that? Do you mean `NaN` in the sense of "not a number"? I don't see how that can happen.

> So, can we call this function `test_passes_y_to_train_split_when_not_nan`? Frankly, I can't invent a better name

With tests, it's always better to use a lengthy name that is more descriptive than a short, less descriptive name. Nobody is going to import your test anyway. If in doubt, tests should also rather contain too many than too few explanatory comments (some skorch tests are hard to understand because we didn't do that right from the start).

Regarding the name itself, I would prefer `test_passes_y_to_train_split_when_not_none` (last word changed), since that is what `y` actually is.

Reply:
> Can you elaborate on that?

I think this is covered in the PR comments. Yes, I just confused `None` and `NaN` in the conversation, sorry. 

I agree to the changes, see the new code.